src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;

```

Comment:
Ok, up to you ;). Your second argments is understandable (a get that those two hooks "before/after batch resolution" with undetermined arguments are huge bags were anything can be put into, making their purpose unclear - too much freedom for serving a purpose that is useful to only one among the Resolvers). The first one, a little bit less since we add 3 (R-dedicated) methods in the PeakResolver as well. Moreover those methods added to a generic interface that is not particularily designed to use R in a systematic way...

But that's ok for me :)